app/ModelFunctions/PhotoFunctions.php

```diff
@@ -286,10 +315,19 @@ public function add(array $file, $albumID_in = 0)
 			}
 		}
 
-		$info = $this->metadataExtractor->extract($path, $mimeType);
+		if ($kind == 'RAW') {
+			$info = $this->metadataExtractor->bare();
+			$this->metadataExtractor->size($info, $path);
+			$this->metadataExtractor->validate($info);
+			$info['type'] = 'raw';
+		} else {
+			$info = $this->metadataExtractor->extract($path, $mimeType);
+		}
```

Comment:
So PHP built-in exif support only works with JPEGs, not RAWs? Bummer... I believe `exiftool` can handle both. We'll need to figure out what to do when we add `exiftool` support...

Reply:
I actually did not try the exif tool on raw files.

It seems that the native one should support it.